.github/workflows/stack.yml

```diff
@@ -5,149 +5,98 @@ on:
+      if: ${{ runner.os == 'Linux' || runner.os == 'macOS' }}
       with:
         path: "~/.stack"
         # A unique cache is used for each stack.yaml.
-        key: ${{ runner.os }}-new-${{ hashFiles(format('stack-{0}.yaml', matrix.ghc-ver)) }}
+        key: ${{ runner.os }}-stack-${{ hashFiles(format('stack-{0}.yaml', matrix.ghc-ver)) }}
 
-    - name: Install needed Ubuntu packages
-      run: |
-        sudo apt-get update && sudo apt-get install libicu-dev
+    - uses: actions/cache@v2
+      name: Cache dependencies (Windows)
+      if: ${{ runner.os == 'Windows' }}
+      with:
+        path: C:\Users\runneradmin\AppData\Roaming\stack
+        key: ${{ runner.os }}-stack-${{ hashFiles(format('stack-{0}.yaml', matrix.ghc-ver)) }}
 
-    - name: Install dependencies for Agda and `agda-tests` (i.e. the test suite).
-      run: |
-        stack build ${ARGS} ${NON_DEFAULT_FLAGS} --test --only-dependencies
+    - name: Update stack package index
+      run: stack update
 
-    - name: Build Agda with the default flags in Agda.cabal. Also build `agda-tests` (i.e. the test suite).
+    - name: Install the icu library (Ubuntu)
+      if: ${{ runner.os == 'Linux' }}
       run: |
-        stack build ${ARGS} --test --no-run-tests
-    - name: Build Agda with the non-default flags Agda.cabal.
-      run: |
-        stack build ${ARGS} ${NON_DEFAULT_FLAGS}
-
-  macOS:
-    if: |
-      !contains(github.event.head_commit.message, '[skip ci]')
-      && !contains(github.event.head_commit.message, '[ci skip]')
-      && !contains(github.event.head_commit.message, '[github skip]')
-      && !contains(github.event.head_commit.message, '[skip github]')
-
-    runs-on: macos-latest
-
-    strategy:
-      matrix:
-        ghc-ver: [8.10.2]
-
-    env:
-      ARGS: "--stack-yaml stack-${{ matrix.ghc-ver }}.yaml --no-terminal --system-ghc --fast"
-      NON_DEFAULT_FLAGS: "--flag Agda:enable-cluster-counting --flag Agda:cpphs --flag Agda:debug"
-
-    steps:
-    - uses: actions/checkout@v2.3.2
-      with:
-        submodules: recursive
+        sudo apt-get update && sudo apt-get install libicu-dev
 
-    - name: Install stack
+    - name: Install the icu librery (Windows)
+      if: ${{ runner.os == 'Windows' }}
       run: |
-        brew install haskell-stack
-        stack update --stack-yaml stack-${{ matrix.ghc-ver }}.yaml --system-ghc
+        stack exec  ${ARGS} -- wget -q http://repo.msys2.org/mingw/x86_64/${ICU_FILE}
+        stack exec  ${ARGS} -- pacman -U --noconfirm ${ICU_FILE}
 
-    - name: Add text-icu path to stack configuration
-      run: |
-        printf "extra-lib-dirs:\n - /usr/local/opt/icu4c/lib\nextra-include-dirs:\n - /usr/local/opt/icu4c/include\n" > ~/.stack/config.yaml
+    - name: Add the icu path to stack configuration (macOS)
+      if: ${{ runner.os == 'macOS' }}
+      run: printf "extra-lib-dirs:\n - /usr/local/opt/icu4c/lib\nextra-include-dirs:\n - /usr/local/opt/icu4c/include\n" > ~/.stack/config.yaml
```

Comment:
Looks like this is how it was before; but I wonder if this line should be `>>` to append instead of replace. Is anything else in that config file? Also noticed that the equivalent line for the cabal config appends.

Reply:
It should be fine for stack. The default one is empty.

Reply:
Not the best practice, I know. I am worried that these two lines may repeat themselves after caching. If the file itself is going to be used as installation instructions, then someone or I have to figure this out. I am happy with it (for now).